tools/abel/validators/README.md

```diff
@@ -0,0 +1,218 @@
+# ABEL Spreadsheet
+
+### Overview
+The ABEL Spreadsheet serves as a frontend for the ABEL application. The
+spreadsheet is the human-friendly interface that allows users to make changes to
+a Building Configuration file.
+
+## Google Sheets template
+Since the format of an ABEL spreadsheet is well-defined, it is suggested that
+you use the following [template](https://docs.google.com/spreadsheets/d/1qKMlpJI5-_h_8innNniEkpatMBcRHSGekrRwTsPQ618/copy#gid=980240783) as either your input or output spreadsheet.
+
+Below is a definition of each column in the ABEL spreadsheet:
+- [Site](#site)
+- [Entities](#entities)
+- [Entity Fields](#entity-fields)
+- [States](#states)
+- [Connections](#connections)
+
+### Site
+
+The `Site` tab holds all information about a project site or building.
+
+`Building Code`
+- String
+- **Required**
+- The human-readable code for a building.
+- e.g. `US-SVL-TC2`
+
+`Entity Guid`
+- String
+- Optional
+- [UUID4](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)) value for a building.
+- **Note:** Unless you are onboarding a Google building, this value is generated by ABEL and does not need to be provided for initial onboarding.
+
+**Note:** A Site should not be considered an entity and should not be listed in
+the `Entities` tab.
+
+### Entities
+
+The `Entities` tab holds all information about reporting and virtual entities in
+a building or site.
+
+**Note:** If ABEL is being used for Google then Guids are required for all
+entities in the `FACILTITIES` namespace.
+
+`Entity Code`
+- String
+- **Required**
+- Human-readable code for an entity.
+- e.g. `AHU-1001`
+
+`Entity Guid`
+- String
+- Optional
+- [UUID4](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)) value for an entity.
+- **Note:** If an entity belongs to the `FACITITLIES` namespace and ABEL is
+  being used or Google, then this field is **required**.
+
+`Etag`
```

Comment:
`used only when updating an existing building config` try to put yourself in the shoes of an MSI.
do I need to fill this or not? is it required for I am doing?